Makefile

```diff
@@ -3,7 +3,7 @@ SDIR=src
 TDIR=pytests
 # compilers
 TSC=./node_modules/typescript/bin/tsc
-TSCFLAGS=-t ES3 --sourceMap --module commonjs --noImplicitAny
```

Comment:
That's because TypeScript will still produce JavaScript files when there are type-checking issues, so it won't recompile. Add `--noEmitOnError`, and you won't have that problem anymore -- if it fails to type check, TypeScript will not produce a JavaScript file.